models/networks/hierarchicalEncoder.py

```diff
@@ -11,10 +11,14 @@
 from utils.dists.distributions import SpikeAndExponentialSmoother
 from utils.dists.MixtureExp import MixtureExp
 from utils.dists.MixtureExpMod import MixtureExpMod
+from utils.dists.GumbelMod import GumbelMod
 
 _SMOOTHER_DICT = {"SpikeExp" : SpikeAndExponentialSmoother, 
-                  "MixtureExp" : MixtureExp}
-_SMOOTHER_MOD_DICT = {"MixtureExp" : MixtureExpMod, "SpikeExp" : None}
+                  "MixtureExp" : MixtureExp,
+                  "Gumbel" : None}
```

Comment:
Is it Gumbel or GumBolt or both? :D Confused!

Reply:
Gumbel is the smoothing procedure of adding Gumbel-Softmax noise and Gumbolt is the application of this procedure to Boltzmann machines.